.github/scripts/common/header_checker.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/env python3
```

Comment:
Would be nice to add a README for these scripts, to provide some documentation for contributors explaining the errors and how to fix them. Could link to it in the error log

Reply:
Can you add the minimum python3 version needed as a comment or in the readme?

Reply:
These are both good points adding now

Reply:
Added a descriptive help message that instructs on how to fix the failing check and added a comment to indicate python version